front/lib/models/assistant/actions/retrieval.ts

```diff
@@ -220,3 +220,224 @@ AgentConfiguration.hasOne(AgentRetrievalConfiguration, {
   foreignKey: { name: "agentId", allowNull: true }, // null = no generation set for this Agent
   onDelete: "CASCADE",
 });
+
+/**
+ * Retrieval Action
+ */
+export class AgentRetrievalAction extends Model<
+  InferAttributes<AgentRetrievalAction>,
+  InferCreationAttributes<AgentRetrievalAction>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare query: string | null;
+  declare relativeTimeFrameDuration: number | null;
+  declare relativeTimeFrameUnit: TimeframeUnit | null;
+  declare topK: number;
+
+  declare retrievalConfigurationId: ForeignKey<
+    AgentRetrievalConfiguration["id"]
+  >;
+}
+AgentRetrievalAction.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    query: {
+      type: DataTypes.TEXT,
+      allowNull: true,
+    },
+    relativeTimeFrameDuration: {
+      type: DataTypes.INTEGER,
+      allowNull: true,
+    },
+    relativeTimeFrameUnit: {
+      type: DataTypes.STRING,
+      allowNull: true,
+    },
+    topK: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+    },
+  },
+  {
+    modelName: "agent_retrieval_action",
+    sequelize: front_sequelize,
+    hooks: {
+      beforeValidate: (retrieval: AgentRetrievalAction) => {
+        // Validation for Timeframe
+        if (
+          retrieval.relativeTimeFrameDuration === null &&
+          retrieval.relativeTimeFrameUnit !== null
+        ) {
+          throw new Error(
+            "Relative time frame must have a duration and unit set or they should both be null"
+          );
+        }
+        if (
+          retrieval.relativeTimeFrameDuration !== null &&
+          retrieval.relativeTimeFrameUnit === null
+        ) {
+          throw new Error(
+            "Relative time frame must have a duration and unit set or they should both be null"
+          );
+        }
+      },
+    },
+  }
+);
+
+AgentRetrievalConfiguration.hasMany(AgentRetrievalAction, {
+  foreignKey: { name: "retrievalConfigurationId", allowNull: false },
+  // We don't want to delete the action when the configuration is deleted
+  // But really we don't want to delete configurations ever.
```

Comment:
Question: we won't allow in the API deletion of agent right? We'll only allow to archive them (so a soft-delete). But still, should I remove some `onDelete: "CASCADE",` in the configuration models?

Reply:
Yes I'm in favor of removing them to prevent deletion as much as possible :+1: